src/doc/model.lisp

```diff
@@ -0,0 +1,303 @@
+;;; The object model for documented values.
+
+(defpackage #:coalton/doc/model
+  (:documentation "Documentation generator for COALTON.")
+  (:use
+   #:cl
+   #:coalton/doc/base
+   #:coalton/doc/environment)
+  (:local-nicknames
+   (#:tc #:coalton-impl/typechecker)
+   (#:entry #:coalton-impl/entry))
+  (:export
+   #:stdlib-packages
+
+   #:constructor-types
+   #:constructors
+   #:context
+   #:documentation
+   #:fields
+   #:instances
+   #:location
+   #:methods
+   #:name
+   #:param-names
+   #:predicate
+   #:type
+   #:tyvars
+
+   #:object-filename
+   #:object-package
+   #:package-entry
+   #:all-entries
+   #:has-entries-p
+   #:value-entries
+   #:type-entries
+   #:class-entries
+
+   #:class-entry
+   #:function-entry
+   #:value-entry
+   #:type-entry
+   #:type-entry-code
+   #:struct-entry
+   #:package-entry-package
+   ))
+
+(in-package #:coalton/doc/model)
+
+(defun stdlib-p (symbol)
+  "A standard library package is any package with the exact name 'coalton' or whose name starts with 'coalton-library'."
+  (let ((name (package-name (symbol-package symbol))))
+    (or (string-equal name "COALTON")
```

Comment:
`coalton-prelude`?

Reply:
At present, the documentation generator organizes symbols by defining file, rather than strictly by package, so adding `coalton-prelude`, at this point, won't generate anything. https://github.com/coalton-lang/coalton/pull/1194 adds support for linking directly to the span in the source file providing the definition of a type/class/value/struct, which will allow a switch to a package rather than file oriented view.